src/components/drawer/drawer-mixin.js

```diff
@@ -0,0 +1,509 @@
+import '@brightspace-ui/core/components/focus-trap/focus-trap.js';
+import '@brightspace-ui/core/helpers/viewport-size.js';
+import { allowBodyScroll, preventBodyScroll } from '@brightspace-ui/core/components/backdrop/backdrop.js';
+import { clearDismissible, setDismissible } from '@brightspace-ui/core/helpers/dismissible.js';
+import { findComposedAncestor, isComposedAncestor } from '@brightspace-ui/core/helpers/dom.js';
+import { forceFocusVisible, getComposedActiveElement, getNextFocusable, tryApplyFocus } from '@brightspace-ui/core/helpers/focus.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { getUniqueId } from '@brightspace-ui/core/helpers/uniqueId.js';
+import { html } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+import { styleMap } from 'lit/directives/style-map.js';
+
+window.D2L = window.D2L || {};
+window.D2L.DrawerMixin = window.D2L.DrawerMixin || {};
+
+const abortAction = 'abort';
+const reduceMotion = matchMedia('(prefers-reduced-motion: reduce)').matches;
+const presetSizes = { 'xs': 150, 'sm': 200, 'md': 300, 'lg': 400, 'xl': 500 };
+
+/**
+ * @fires d2l-drawer-close
+ * @fires d2l-drawer-open
+ */
+export const DrawerMixin = superclass => class extends RtlMixin(superclass) {
+
+	static get properties() {
+		return {
+			/**
+             * Aria label for drawer component
+             */
+			label: { type: String },
+			/**
+             * Whether or not the drawer is open (true if drawer is open)
+             */
+			open: { type: Boolean, reflect: true },
+			/**
+             * Drawer body positiong (top, right, bottom, left)
+             */
+			position: { type: String, reflect: true },
+			_closeAborted: { type: Boolean }, // Still working on how to animate - may not need
+			_nestedShowing: { type: Boolean },
+			_overflowBottom: { type: Boolean },
+			_overflowTop: { type: Boolean },
+			_parentDrawer: { type: Object },
+			_scroll: { type: Boolean },
+			_state: { type: String, reflect: true },
+		};
+	}
+
+	constructor() {
+		super();
+		this.label = 'Drawer';
+		this.open = false;
+		this.position = 'right';
+		this._closeAborted = false;
+		this._drawerId = getUniqueId();
+		this._nestedShowing = false;
+		this._overflowBottom = false;
+		this._overflowTop = false;
+		this._parentDrawer = null;
+		this._scroll = false;
+		this._state = null;
+		this._updateSize = this._updateSize.bind(this);
+		this._updateOverflow = this._updateOverflow.bind(this);
+	}
+
+	async updated(changedProperties) {
+		super.updated(changedProperties);
+		if (!changedProperties.has('open')) return;
+
+		if (this.open) {
+			this._open();
+		} else {
+			this._close();
+		}
+	}
+
+	close() {
+		if (!this.open) return;
+		this.open = false;
+		return new Promise(resolve => {
+			setTimeout(async() => {
```

Comment:
What's the reason for wrapping these in `setTimeout` calls?

Reply:
stemmed from a copy and paste misnomer - `resize` on dialog-mixin.js in core required it for timing purposes in Firefox https://github.com/BrightspaceUI/core/pull/2233.
Seems related to use of native dialog component though and therefore should have no influence on drawer - will remove.